python/pyspark/sql/session.py

```diff
@@ -611,8 +611,8 @@ def _inferSchema(
         :class:`pyspark.sql.types.StructType`
         """
         first = rdd.first()
-        if not first:
-            raise ValueError("The first row in RDD is empty, " "can not infer schema")
+        if first is None:
+            raise ValueError("The first row in RDD is empty, can not infer schema")
```

Comment:
Sorry I was confused with `DataFrame.take`. `RDD.take` seems like always retuning something because it's `items: List[T] = []`. Can we just check if the length is 0 instead?